### PR TITLE
Add support for WPT style test-wait class name

### PR DESCRIPTION
**Pull request by rniwa**
#### b4b50d4068c80ce1c9db8c90717b72a9046674e1
<pre>
Add support for WPT style test-wait class name
<a href="https://bugs.webkit.org/show_bug.cgi?id=242743">https://bugs.webkit.org/show_bug.cgi?id=242743</a>

Reviewed by Darin Adler and Simon Fraser.

This patch adds the support for test-wait class on the document element to prevent the test from
completion much like <a href="https://commits.webkit.org/229177@main">https://commits.webkit.org/229177@main</a>, which added the support for reftest-wait.

* LayoutTests/fast/harness/test-wait-expected.txt: Added.
* LayoutTests/fast/harness/test-wait.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/crashtests/multiple-references-id-crash-002-expected.txt:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/filter-effects/crashtests/multiple-references-id-crash-002-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/filter-effects/crashtests/multiple-references-id-crash-002-expected.txt:

* Tools/DumpRenderTree/DumpRenderTree.xcodeproj/project.pbxproj:
* Tools/DumpRenderTree/mac/FrameLoadDelegate.mm:
(-[FrameLoadDelegate dumpAfterWaitAttributeIsRemoved:]):
* Tools/DumpRenderTree/win/FrameLoadDelegate.cpp:
(dumpAfterWaitAttributeIsRemoved):
* Tools/TestRunnerShared/CMakeLists.txt:
* Tools/TestRunnerShared/WPTFunctions.cpp: Renamed from Tools/TestRunnerShared/ReftestFunctions.cpp.
(WTR::sendTestRenderedEvent):
(WTR::hasTestWaitAttribute): Renamed from hasReftestWaitAttribute and added the check for test-wait.
* Tools/TestRunnerShared/WPTFunctions.h: Renamed from Tools/TestRunnerShared/ReftestFunctions.h.
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::hasTestWaitAttribute): Renamed from hasReftestWaitAttribute.
(WTR::dumpAfterWaitAttributeIsRemoved):
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/252487@main">https://commits.webkit.org/252487@main</a>
</pre>

### CODE REVIEW

**Review by darinadler — APPROVED** [review]
No progressions in WPT from adding this?

**Comment by rniwa** [comment]
> No progressions in WPT from adding this?

I'm sure there are some progressions! I just haven't checked them yet.

File: LayoutTests/fast/harness/test-wait.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+<div id='result'>FAIL</div>
+<script>
+if (window.testRunner)
+    testRunner.dumpAsText();
+
+onload = () => {
+    setTimeout(() => {
+        result.textContent = 'PASS';
+        document.documentElement.removeAttribute('class');
+    }, 100);
```
Comment (smfr) [inline LayoutTests/fast/harness/test-wait.html]: Does this have to be as long as 100ms?. That's a pretty slow test.

Comment (smfr) [reply LayoutTests/fast/harness/test-wait.html]: You could do something like have two requestAnimationFrames, remove `test-wait` in the first one, and make sure the second one's log doesn't show in the results.

**Review by smfr — APPROVED** [review]

File: LayoutTests/fast/harness/test-wait.html (hunk 1 above)
Comment (rniwa) [reply LayoutTests/fast/harness/test-wait.html]: oh, I was just using 100ms as a safe time. I reduced it to 10ms.

**Pull request merged**